### ISSUE TITLE
Support macOS for deployments

### ISSUE THREAD

**Issue by iMicknl** [Azure-Samples/azure-search-knowledge-mining#9]
### This issue is for a:
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
AzureRM doesn't support macOS.

### Expected/desired behavior
MacOS platform supported as option for deployments, which can be achieved by:

- Rewrite Powershell script to use [`Az module` instead of `AzureRM module`](https://azure.microsoft.com/nl-nl/blog/azure-powershell-cross-platform-az-module-replacing-azurerm/). 
- Add a Bash deployment option uzing AZ-CLI.

**Comment by iMicknl** [comment]
Could be solved temporarily by adding `Enable-AzureRmAlias` (session only) to the script. 
[Migrate Azure PowerShell from AzureRM to Az](https://docs.microsoft.com/en-us/powershell/azure/migrate-from-azurerm-to-az?view=azps-2.2.0)

**Comment by iMicknl** [comment]
Fixed in #14.
